R/ncdf.R

```diff
@@ -506,30 +512,43 @@ read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(
   return(raster)
 }
 
-.get_nc_dimensions <- function(dimensions, coord_var, coords, nc, dims, 
-                               var_names, curvilinear, eps, ignore_bounds) {
-  
+.get_nc_dimensions <- function(dimensions, coord_var, coords, nc, dims,
+                               var_names, curvilinear, eps, ignore_bounds,
+                               atts) {
+
   to_rectilinear = FALSE
+
+  ## hack longitudes for #277
+  if (any(lon_coord <- grepl("lon", coord_var$X, ignore.case = TRUE))) {
+    lon_coord <- coord_var$X[lon_coord][1]
+    lons <- coords[[lon_coord]]
+    if (!is.null(lons) && !regular_intervals(lons, epsilon = eps) &&
+        .is_degrees(atts, lon_coord) &&
```

Comment:
I factored out a .is_degrees function and call it here now.

Reply:
LGTM!